src/electrum_extended_priv_key.rs

```diff
@@ -58,6 +58,37 @@ impl ElectrumExtendedPrivKey {
     pub fn xprv(&self) -> &ExtendedPrivKey {
         &self.xprv
     }
+
+    /// converts to electrum format
+    pub fn electrum_xprv(&self) -> Result<String, String> {
+        let data = match (self.xprv.network, self.kind.as_str()) {
+            (Network::Testnet, "pkh") => [0x04u8, 0x35, 0x83, 0x94], // tprv
+            (Network::Testnet, "sh(wpkh") => [0x04u8, 0x4a, 0x4e, 0x28], // uprv
+            (Network::Testnet, "sh(wsh") => [0x02u8, 0x42, 0x85, 0xb5], // Uprv
+            (Network::Testnet, "wpkh") => [0x04u8, 0x5f, 0x18, 0xbc], // vprv
+            (Network::Testnet, "wsh") => [0x02u8, 0x57, 0x50, 0x48], // Vprv
+            (Network::Bitcoin, "pkh") => [0x04u8, 0x88, 0xad, 0xE4], // xprv
+            (Network::Bitcoin, "sh(wpkh") => [0x04u8, 0x9d, 0x78, 0x78], // yprv
+            (Network::Bitcoin, "sh(wsh") => [0x02u8, 0x95, 0xb0, 0x05], // Yprv
+            (Network::Bitcoin, "wpkh") => [0x04u8, 0xb2, 0x43, 0x0c], // zprv
+            (Network::Bitcoin, "wsh") => [0x02u8, 0xaa, 0x7a, 0x99], // Zprv
+            _ => { return Err("unknown type".to_string()); },
+        };
+        let mut data = Vec::from(data);
+        data.push(self.xprv.depth);
+        data.extend(self.xprv.parent_fingerprint.as_bytes());
+        let child_number: u32 = self.xprv.child_number.into();
+        data.extend(child_number.to_be_bytes());
+        data.extend(self.xprv.chain_code.as_bytes());
+        data.push(0u8);
```

Comment:
Not entirely sure yet about this byte. Still needs verification and more testing.

Reply:
Any update on this? I am not sure either and it's the reason I've still doubt to merge this PR

Reply:
All tests I did so far were positive. 
Electrum also just inserts a zero there without an explanation:
https://github.com/spesmilo/electrum/blob/master/electrum/bip32.py#L183
In https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#serialization-format there is an explicit 0x00 padding byte listed for private keys.
So I'm positive, that this is good.

Reply:
LGTM